client/src/http.rs

```diff
@@ -13,7 +14,7 @@ pub use http::{Method, Response, StatusCode};
 pub trait RequestBuilder {
     /// Create a new builder with specified method and URL. Entrypoint for most client operations.
     #[must_use]
-    fn new(method: Method, url: impl AsRef<str>) -> Self;
+    fn new(method: Method, url: Url) -> Self;
```

Comment:
It is kind of a breaking change. When builder is used for different HTTP-client crates, they might have their own abstractions over URL, and this might lead to extra allocations.

Though I don't think it is critical, since the whole client API is going to be drastically changed:

- #3130